### PR TITLE
Project-todos Elle Frankenberg

### PR DESCRIPTION
**Pull request by ElleFrankenberg**
Link to my deployed project:
https://boring-booth-3f759d.netlify.app/

My reflections on how this week's project turned out:

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: code/README (kopia).md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Project Todos
+
+In this week's project was to build a to-do app using react/redux.
+We were to structure a redux store, write reducers and actions and interact with our store using selectors and dispatching actions.
+
+## The problem
+
+I started by sketching out what my app would look like, but didn´t come up with a design that i liked. So instead I found a design on canva.com and tried to imitate that, but also add something own to it. In a previous project, I had trouble understanding how to add fonts, but this week I figured it out, so I'm happy about that :)
+Every week I find the projects very difficult (this week as well) so I think it has become important for me to be happy about all the small steps I take in my progress.
```
Comment (TobiasHagberg) [inline code/README (kopia).md:9]: The small wins all add up to something great :) It is really great that you "see" and can appreciate those moments!

File: code/README (kopia).md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+We were to structure a redux store, write reducers and actions and interact with our store using selectors and dispatching actions.
+
+## The problem
+
+I started by sketching out what my app would look like, but didn´t come up with a design that i liked. So instead I found a design on canva.com and tried to imitate that, but also add something own to it. In a previous project, I had trouble understanding how to add fonts, but this week I figured it out, so I'm happy about that :)
+Every week I find the projects very difficult (this week as well) so I think it has become important for me to be happy about all the small steps I take in my progress. 
+Previous I´v also found styling a bit tricky, especially inputs. This week I put more time into it, and I'm pretty happy with the result. It was also fun to understand how to make my own svgs in Illustrator and how to add them to my project. But I have not yet figured out how to get them into a checkbox properly, so this week I "cheated" a bit and removed the box from the checkbox. But design-wise i think i works.. However, I understand that my design this week is not very user-friendly.
```
Comment (TobiasHagberg) [inline code/README (kopia).md:10]: All the extra effort really shows, great ambition and work!

File: code/README (kopia).md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+I think I have become better at googling and finding solutions to my problems, I have discovered that I understand more in the articles that I find and what solutions to try.
```
Comment (TobiasHagberg) [inline code/README (kopia).md:11]: This is so true and the path to understanding code other than your own becomes shorter for each week :)

File: code/README (kopia).md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+If I had more time it would have been nice to do a transition on my front page, but i could only get the transition to work one way, so i removed it.
+And also use "draggable" which I downloaded to be able to move my to-dos around on the page. 
+
+I noticed when I deployed my page on netlify that in safari my checkbox disappears, this I would have fixed if I had had more time...
```
Comment (TobiasHagberg) [inline code/README (kopia).md:16]: I viewed this on my iPhone and in safari and this was the only comment I had. Besides this it is all 🙌

File: code/package.json [hunk 5]
```diff
@@ -12,7 +13,11 @@
     "eslint-plugin-react-hooks": "^2.3.0",
     "react": "^16.12.0",
     "react-dom": "^16.12.0",
-    "react-scripts": "3.2.0"
+    "react-draggable": "^4.4.3",
```
Comment (TobiasHagberg) [inline code/package.json:16]: Noticed your comment above but this would have been really cool to see

File: code/src/App.js [hunk 6]
```diff
@@ -1,9 +1,24 @@
 import React from 'react'
+import GlobalFonts from './fonts/fonts'
```
Comment (TobiasHagberg) [inline code/src/App.js:2]: I am curious, you will have to explain this to me one day!

File: code/src/SVGs/arrow.svg [hunk 7]
```diff
@@ -0,0 +1 @@
+<svg id="pil" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 168.41 231.61"><defs><style>.cls-1{fill:none;stroke:#fe5bb1;stroke-linecap:square;stroke-miterlimit:10;stroke-width:15px;}</style></defs><polyline class="cls-1" points="157.93 10.48 12.77 115.92 157.93 221.14"/></svg>
```
Comment (TobiasHagberg) [inline code/src/SVGs/arrow.svg:1]: Really nice touch with your own svgs

File: code/src/components/Header.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+const Header = () => {
+  const [activeHeader, setActiveHeader] = useState('false')
+  const [activeIcon, setActiveIcon] = useState('false')
+  const [activeText, setActiveText] = useState('false')
+
+  const handleToggle = () => {
+    setActiveHeader(!activeHeader)
+    setActiveIcon(prevMode => !prevMode)
+    setActiveText(prevMode => !prevMode)
```
Comment (TobiasHagberg) [inline code/src/components/Header.js:15]: I still find conditional rendering a bit tricky, this is impressive! :)

File: code/src/fonts/fonts.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import { createGlobalStyle } from 'styled-components'
```
Comment (TobiasHagberg) [inline code/src/fonts/fonts.js:1]: Again, new concept for me and would like it explained one day.

File: code/src/index.css [hunk 10]
```diff
@@ -5,9 +5,250 @@ body {
...
   -moz-osx-font-smoothing: grayscale;
+
+  background-color:  #fe5bb1;
 }
 
 code {
   font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
     monospace;
 }
+
```
Comment (TobiasHagberg) [inline code/src/index.css:16]: Wow this is both ambitious and really well done! The whole css file that is :)

**Review by TobiasHagberg — APPROVED** [review]
This is such a nice project Elle! Really well done and I especially appreciate the determination you express in the readme and the direction you have taken this project in. It is something different and it is really good!

File: code/src/components/TodoList.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+          <input
+            type="checkbox"
+            checked={task.isComplete}
+            onChange={() => dispatch(todos.actions.toggleComplete(task.id))}
+          />
+          <p className="todo-text">{task.description}</p>
+          <button className="delete-button" onClick={() => dispatch(todos.actions.removeTask(task.id))}>
+            Delete -
+          </button>
+        </div>
```
Comment (KarolCislowski) [inline code/src/components/TodoList.js:33]: I think it should be a separate component 😉

File: code/src/index.css [hunk 12]
```diff
@@ -5,9 +5,250 @@ body {
...
+  display: flex;
+  align-items:center;
+  justify-content: center;
+  width: 30px;
+  height: 30px;
+}
+
+input[type="checkbox"]:checked:after {
+  content: url('./SVGs/checked.svg');
+}
```
Comment (KarolCislowski) [inline code/src/index.css:130]: Nice 😻

**Review by KarolCislowski — APPROVED** [review]
Very nice design and great code 🥇

**Pull request closed without merging**